### PR TITLE
gocollector: Add regex option to allow collection of debug runtime metrics

### PR DESCRIPTION
**Pull request by ArthurSens**
At the same time, it also solves the linting problem we're getting on main

### CODE REVIEW

**Comment by kakkoyun** [comment]
> This is a quick easy fix, but maybe we could extend the Go collector to allowList debug metrics instead(Consequentially we'd start using the `withDebugMetrics` method)?

👍

File: prometheus/collectors/go_collector_go121_test.go [hunk 1]
```diff
@@ -149,26 +149,3 @@ func withSchedulerMetrics() []string {
 		"go_threads",
 	}
 }
-
-func withDebugMetrics() []string {
```
Comment (kakkoyun) [inline prometheus/collectors/go_collector_go121_test.go:153]: Let's utilize them, in a unit test rather than removing them.

**Comment by kakkoyun** [comment]
@ArthurSens, shall we revive this?

**Comment by ArthurSens** [comment]
Yeah, it's on my list for quite a while now and I don't even remember what blocked me 😓 

I'll try to get back to this at some point this week

**Comment by ArthurSens** [comment]
Alright, finally found time to come back to this :) 

I had to modify the approach a bit after #1476. I'm now generating the `withDebugMetrics()` function with the script that @SachinSahu431 created. 

There aren't any debug metrics for older debug metrics, so I've updated the script to also generate the function even if empty.

File: prometheus/collectors/go_collector_go120_test.go [hunk 2]
```diff
@@ -33,6 +30,9 @@ func withAllMetrics() []string {
 		"go_cpu_classes_scavenge_total_cpu_seconds_total",
 		"go_cpu_classes_total_cpu_seconds_total",
 		"go_cpu_classes_user_cpu_seconds_total",
+		"go_gc_cycles_automatic_gc_cycles_total",
+		"go_gc_cycles_forced_gc_cycles_total",
+		"go_gc_cycles_total_gc_cycles_total",
```
Comment (ArthurSens) [inline prometheus/collectors/go_collector_go120_test.go:35]: After running the script, the order changed 😬

Comment (SachinSahu431) [reply prometheus/collectors/go_collector_go120_test.go:35]: The metrics were not sorted in the previous files

https://github.com/prometheus/client_golang/pull/1483#discussion_r1545534373

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**